parsec/serde/schema.py

```diff
@@ -99,17 +100,27 @@ def get_obj_type(self, obj):
 
     def _get_schema(self, type):
         if self._instantiated_schemas is None:
+            # Assert all schemas key are Enum:
+            if not all([isinstance(k, Enum) for k in self.type_schemas]):
+                raise ValueError("type_schemas key can only be an Enum")
             self._instantiated_schemas = {
                 k: v if isinstance(v, Schema) else v() for k, v in self.type_schemas.items()
             }
+
             if self.fallback_type_schema and not isinstance(self.fallback_type_schema, Schema):
                 self._instantiated_fallback_schema = self.fallback_type_schema()
 
+        if not isinstance(type, Enum):
```

Comment:
If type can be Enum or str, we should add typing to the method signature

Reply:
done